sources/packages/backend/libs/integrations/src/cas/cas-formaters.ts

```diff
@@ -0,0 +1,51 @@
+import { convertToASCII } from "@sims/utilities";
+
+const CAS_SUPPLIER_NAME_MAX_LENGTH = 80;
+const CAS_ADDRESS_MAX_LENGTH = 35;
+const CAS_CITY_MAX_LENGTH = 25;
+
+/**
+ * Format the full name in the expected format (last name, given names).
+ * Ensure only ASCII characters are present, make all upper case,
+ * and enforce the maximum length accepted by CAS.
+ * @param firstName first name (given names).
+ * @param lastName last name.
+ * @returns formatted full name.
+ */
+export function formatUserName(firstName: string, lastName: string): string {
+  const formattedName = `${lastName}, ${firstName}`.substring(
+    0,
+    CAS_SUPPLIER_NAME_MAX_LENGTH,
+  );
+  return convertToASCII(formattedName).toUpperCase();
+}
+
+/**
+ * Ensure only ASCII characters are present, make all upper case,
+ * and enforce the maximum length accepted by CAS.
+ * @param address address to be formatted.
+ * @returns formatted address.
+ */
+export function formatAddress(address: string): string {
+  return convertToASCII(
+    address.substring(0, CAS_ADDRESS_MAX_LENGTH),
+  ).toUpperCase();
+}
+
+/**
+ * Ensure city has the max length expected by CAS.
+ * @param address city to be formatted.
+ * @returns formatted city.
+ */
+export function formatCity(city: string): string {
+  return city.substring(0, CAS_CITY_MAX_LENGTH);
```

Comment:
Is this not expected to be passed as uppercase?

Reply:
The AC states "City: Use as-is" hence not a requirement.